### PR TITLE
Add CHANGELOG for 1.5.1

### PR DESCRIPTION
**Pull request by beta-ziliani**
Preview: https://github.com/beta-ziliani/crystal/blob/release/1.5.1/CHANGELOG.md

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,218 @@
...
+
+- Fix `Class#nilable?` for recursive unions and root types ([#12353](https://github.com/crystal-lang/crystal/pull/12353), thanks @HertzDevil)
+
+### Numeric
+
+- Fix `Float#abs` and `Number#format` for `-0.0` ([#12424](https://github.com/crystal-lang/crystal/pull/12424), thanks @straight-shoota)
+
+### Text
+
+- Fix null dereference when passing empty slice to Base64.encode ([#12377](https://github.com/crystal-lang/crystal/pull/12377), thanks @dscottboggs)
```
Comment (straight-shoota) [inline CHANGELOG.md:13]:
```suggestion
- Fix null dereference when passing empty slice to `Base64.encode` ([#12377](https://github.com/crystal-lang/crystal/pull/12377), thanks @dscottboggs)
```

**Review by straight-shoota — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,219 @@
...
+
+- Formatter: Fix assign followed by comment ([#12319](https://github.com/crystal-lang/crystal/pull/12319), thanks @straight-shoota)
+- Handle formatting annotated method parameters ([#12446](https://github.com/crystal-lang/crystal/pull/12446), thanks @Blacksmoke16)
+
+## Other
+
+- Update distribution-scripts ([#12359](https://github.com/crystal-lang/crystal/pull/12359), thanks @straight-shoota)
+- Update distribution-scripts ([#12333](https://github.com/crystal-lang/crystal/pull/12333), thanks @straight-shoota)
+
+# 1.5.0 (2022-07-06)
```
Comment (Blacksmoke16) [inline CHANGELOG.md:40]: Should this PR have the changelog for `1.5.0` in it?

Comment (straight-shoota) [reply CHANGELOG.md:40]: Apparently the release commit for 1.5.0 is missing from the `release/1.5` branch 😮

Comment (beta-ziliani) [reply CHANGELOG.md:40]: indeed, this is why this one started with that lost commit

**Pull request closed without merging**